pkg/karmadactl/cmdinit/karmada/check.go

```diff
@@ -0,0 +1,37 @@
+package karmada
+
+import (
+	"context"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog/v2"
+	apiregistrationv1 "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
+	aggregator "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset"
+)
+
+func waitAPIServiceReady(c *aggregator.Clientset, name string, timeout int) error {
+	if err := wait.PollImmediate(time.Second, time.Duration(timeout)*time.Second, func() (done bool, err error) {
+		apiService, e := c.ApiregistrationV1().APIServices().Get(context.TODO(), name, metav1.GetOptions{})
+		if e != nil {
+			return false, nil
+		}
+		for _, condition := range apiService.Status.Conditions {
+			if condition.Type == apiregistrationv1.Available {
+				if condition.Status == apiregistrationv1.ConditionTrue {
```

Comment:
Can we use the [IsAPIServiceConditionTrue](https://github.com/kubernetes/kubernetes/blob/52cd4d53ac8ac33cacd3d2c618cf22f1e3fd6a7e/staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/v1/helper/helpers.go#L125)?